digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/RoadNameService.scala

```diff
@@ -1,29 +1,102 @@
 package fi.liikennevirasto.viite
 
 import fi.liikennevirasto.digiroad2.oracle.OracleDatabase
-import fi.liikennevirasto.viite.dao.{RoadName, RoadNameDAO}
+import fi.liikennevirasto.digiroad2.user.User
+import fi.liikennevirasto.viite.dao.{ProjectLinkNameDAO, RoadName, RoadNameDAO}
 import org.joda.time.DateTime
 import org.slf4j.LoggerFactory
 
 import scala.util.control.NonFatal
 
+case class RoadNameRows(roadId: Long, editions: Seq[RoadNameEditions])
+
+case class RoadNameEditions(editedField: String, value: String)
```

Comment:
We shouldn't use it in this way because it's not maintainable. We should have a field for each editable field.